File: swarm_input/linked_list.rb
class Node
  attr_accessor :value, :next_node
  def initialize(value); @value = value; @next_node = nil; end
end

class LinkedList
  def initialize; @head = nil; end
  def prepend(value); node = Node.new(value); node.next_node = @head; @head = node; end
  def to_a
    result, current = [], @head
    while current; result << current.value; current = current.next_node; end
    result
  end
end

ll = LinkedList.new
[3, 2, 1].each { |v| ll.prepend(v) }
puts ll.to_a.inspect

File: swarm_input/linked_list.py
class Node:
    def __init__(self, value):
        self.value = value
        self.next_node = None

class LinkedList:
    def __init__(self):
        self.head = None

    def prepend(self, value):
        node = Node(value)
        node.next_node = self.head
        self.head = node

    def to_a(self):
        result = []
        current = self.head
        while current:
            result.append(current.value)
            current = current.next_node
        return result

ll = LinkedList()
for v in [3, 2, 1]:
    ll.prepend(v)
print(ll.to_a())
